modules/internal/ChapelArray.chpl

```diff
@@ -768,8 +768,11 @@ module ChapelArray {
       // otherwise the parentDom gets destroyed in the sparse case; see
       // sparse/bradc/CSR/sparse.chpl as an example
       //
-      pragma "no auto destroy" var dom: domainType;
-      return chpl__buildSparseDomainRuntimeType(d, dom._value.parentDom);
+      proc getParentDomType() type {
+        var dom : domainType;
+        return __primitive("static typeof", dom._value.parentDom.type);
+      }
+      return chpl__buildSparseDomainRuntimeType(d, __primitive("get runtime type field", getParentDomType(), domainType, "dom"));
```

Comment:
It might be nice to store the result of this primitive in a temporary. Why? Two reasons: 1) make the line shorter and 2) rely less upon how normalize adds a temporary in this particular case.